Code/Editor/EditorViewportWidget.cpp

```diff
@@ -103,7 +103,7 @@
 AZ_CVAR(
     bool, ed_visibility_logTiming, false, nullptr, AZ::ConsoleFunctorFlags::Null, "Output the timing of the new IVisibilitySystem query");
 AZ_CVAR(bool, ed_useNewCameraSystem, true, nullptr, AZ::ConsoleFunctorFlags::Null, "Use the new Editor camera system");
-AZ_CVAR(bool, ed_showCursorCameraLook, false, nullptr, AZ::ConsoleFunctorFlags::Null, "Show the cursor when using free look with the new camera system");
+AZ_CVAR(bool, ed_showCursorCameraLook, true, nullptr, AZ::ConsoleFunctorFlags::Null, "Show the cursor when using free look with the new camera system");
```

Comment:
is this caused by a recent change? or not a 100% repo issue?
i tried locally and when i rclick to rotate i'm not getting a context menu (i'm maybe 1.5days out of date from latest dev).

if its not 100% repro should we be making this change?

Reply:
This is super minor and is negatively impacting some people (it might depend on mouse dpi sensitivity among other things) - I'd rather just disable the cursor hiding for now and then work on a proper fix rather than make people wait (having the context menu constantly pop-up is a massive pain and it is a pain to have to let everyone know about this flag (you can always disable it again locally if you want))

Reply:
Sorry, just trying to understand the change in the scope of things.
i find it odd that the visibility of the cursor affects the display of the rclick context menu (though i'm not super familiar with this area of code).

Reply:
No worries, sorry it is a bit of a tricky one for a couple of reasons.

The primary reason hiding the cursor is a problem is if we don't, we effectively keep the cursor in the position it was when RMB down happened (even when you've looked around a bunch by physically moving the mouse). In `EditorContextMenu.cpp` we listen for a mouse up event and check if the position of the mouse after a mouse up has a large enough delta from mouse down to warrant showing the menu or not (if the mouse hasn't moved, we show the menu, if it has, we don't). The problem is the cursor position we get in mouse up is the same as RMB down (as when we capture the cursor as far as the app is concerned, it doesn't move).

Before we used to listen for mouse move events, and update the mouse position in `EditorContextMenuUpdate`, but that also had a problem because when you're using RMB to do a camera look, it gets the event first, and then swallows the move event after it activates (until you release RMB again) as nothing else should get that event (it's being used by the look). So we can't reliable listen for mouse move events there either.

I think the 'correct' fix is to detect raw mouse movement events (measure in units called 'mickey' (https://www.webopedia.com/definitions/mickey/)) and prevent the context menu from appearing once enough of a movement has been detected. I might change my mind about this but I think it is going to be the most reliable mechanism.

In the meantime if we just keep showing the cursor and don't capture it, all this logic works fine 😛 So in the meantime people will have to put up with seeing the cursor but this context menu appearing won't happen (which can get suuuuuper frustrating as Chris can attest!).

I hope that makes some sense, once I'm done with this Viewport Ui focus issue I'll be looking at fixing this properly 👍

Reply:
cheers for the insight! things like this can be tricky.